saleor/order/utils.py

```diff
@@ -338,3 +343,18 @@ def get_voucher_discount_for_order(order: Order) -> Money:
 
 def match_orders_with_new_user(user: User) -> None:
     Order.objects.confirmed().filter(user_email=user.email, user=None).update(user=user)
+
+
+def _get_unique_pdf_file_name():
+    while True:
+        pdf_path = f"{settings.MEDIA_ROOT}/{token_urlsafe(32)}.pdf"
+        if not os.path.isfile(pdf_path):
+            return pdf_path
```

Comment:
Hmmm what about UUID4 rather than this O(N) loop?

Reply:
This will also fail if we decide not to use local media storage, i.e. an instance storage gets wiped

Reply:
Yeah I guess we could just UUID as we do for example for checkout/order tokens.